File: src/main/java/io/sommers/aiintheipaw/model/channel/Channel.java
package io.sommers.aiintheipaw.model.channel;

import io.sommers.aiintheipaw.model.service.IService;
import io.vavr.control.Option;
import jakarta.annotation.Nullable;

public record Channel(
        long id,
        IService service,
        @Nullable String guildId,
        String channelId
) implements IChannel {

    @Override
    public long getId() {
        return this.id();
    }

    @Override
    public IService getService() {
        return this.service();
    }

    @Override
    public Option<String> getGuildId() {
        return Option.of(this.guildId());
    }

    @Override
    public String getChannelId() {
        return this.channelId();
    }
}

File: src/main/scala/model/channel/Channel.scala
package io.sommers.aiintheipaw
package model.channel

import model.guild.Guild
import model.service.Service

trait Channel {
  val id: Long

  val service: Service

  val guild: Guild

  val channelId: String

  val displayName: String
}

case class ChannelImpl(
  override val id: Long,
  override val channelId: String,
  override val service: Service,
  override val guild: Guild,
  override val displayName: String
) extends Channel

